Improve documentation (#187)
openpharma/clinsight: vignettes/clinsight.Rmd

---
title: "Introduction to ClinSight"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction to ClinSight}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup, echo=FALSE}
library(clinsight)
```

> ## Test ClinSight

Author: Probably not indeed; removed

To get started, you can easily run and test ClinSight by using the function `clinsight::test_clinsight()`. This function loads all data in a temporary folder that will be cleaned up after use. Note that this function is only meant for testing and development, and is a wrapper around `clinsight::run_app()`, the main function of this package. `clinsight::run_app()` also works, but creates databases in the working directory that will save user actions such as review status of study data, which is needed for deployment in a production environment. 

```{r eval=FALSE}
# Not run:
# If evaluated, this will run ClinSight with internal test data
test_clinsight()
```

> The start screen will look like the following:

Reviewer: ```suggestion
Once launched, the `Start` screen will look like the following:
```

![](images/clinsight-start_screen.png){width="556"}

## Create custom data and metadata

> In order to use custom study data with ClinSight, the raw custom data needs to fulfill few specific requirements. Secondly, a ClinSight metadata file needs to be customized and created based on the custom data (see `create_clinsight_metadata()` and `get_metadata()`. If these requirements are met, the data can be merged with the study-specific `metadata` to successfully create custom `ClinSight`-compatible study data. All details about the metadata template can be found in `vignette("Metadata")`.

Author: Applied with changes

### Raw data structure

> Raw study data needs to needs to be in long format in order to use it with ClinSight, with the columns being in character vector format. The following columns are expected to be in the raw data.

Reviewer: ```suggestion
Raw study data needs to needs to be in "long format" in order to ingest it within the ClinSight application, with the all columns being in character vector format. The following columns are expected in your raw data:
```

Author: Adapted and adjusted. Also added a sentence about the variable names being customizable

| subject_id     | Unique identifier for a subject                                                                                |
| event_id       | Identifies events within a study, such as study visits.                                                        |
| event_date     | The date of the event. Should be convertible from a character vector to a valid date within R.                 |
| event_repeat   | Counter containing a sequence number that increments if an event_id is repeated within the same subject.       |
| form_id        | Identifier for each form. Can be mostly ignored, since new form identifiers are set within ClinSight metadata. |
| form_repeat    | Counter containing a sequence number that increments if a form is repeated within the same subject.            |
| var            | Name of the variable to use in ClinSight. Longer/more appropriate names will be specified in metadata.         |
| item_value     | Value of the variable (`var`) in question.                                                                     |
| edit_date_time | The time at which the value was last edited. Important for efficient reviewing of the data within ClinSight.   |

> As an example, we will use raw data included in the package. Note that if the column names differ, they can be adjusted in the metadata if needed, so that they will show up as in the table above after merging.

Reviewer: ```suggestion
To furnish an example, we will use the raw data pre-installed with the package. Note that if your column names differ, they can be adjusted (renamed) using the first tab from the metadata file as needed.
```

```{r}
raw_data <- get_raw_csv_data(system.file("raw_data", package = "clinsight"))
head(raw_data)
```

### Merge and prepare data

> When metadata is adjusted properly to the study-specific needs (see `vignette("Metadata")`), create an R object from the Excel file:

Author: implemented with adaptation. Removed the dangling modifier

```{r }
meta_custom <- get_metadata(system.file("metadata.xlsx", package = "clinsight"))
```

> Then, merge the metadata with the raw data. This will give some information and will try to throw an informative error if an incompatible change was made:

Reviewer: ```suggestion
Then, you will need to merge the metadata object with the raw data. This will output an informative error message if an incompatible change was made:
```

Author: Implemented

```{r}
merged_study_data <- merge_meta_with_data(raw_data, meta_custom)
```

> Finally, we can start ClinSight with our custom data and metadata:

Reviewer: ```suggestion
Finally, we can test launch ClinSight with our custom data and metadata:
```

Author: Implemented locally, commit will follow

```{r, eval=FALSE}
#Not run
test_clinsight(merged_study_data, meta_custom)
```

> You will notice that the application shows a warning if some variables defined in the metadata are not available in the actual data, similar to the message below. It is useful to check this output carefully to see if all variable names were entered correctly.

Reviewer: ```suggestion
Notice that the application shows a warning if some variables defined in the metadata are not available in the actual data, similar to the message below. It is useful to check this output carefully to see if all variable names were entered correctly (or incorrectly).
```

Author: Implemented locally with slight adjustment. Commit will follow

```{r}
# Warning in check_appdata(app_data, meta) :
#   Not all variables defined in metadata are present in the data.
#              var               item_name
# 1 AE_AESER_AEOUT             SAE outcome
# 2     WHO_WHOCAT   WHO.subclassification
# 3       DMOD_DAT    DoseModificationDate
# 4      DMOD_REAS  DoseModificationReason
# 5      DMOD_DOSE DoseModificationNewDose
```

To improve and adjust the application, simply stop the application, adjust the metadata, merge it again with the data, and try again. For example, if we want to change the name of 'Systolic blood pressure' to 'SBP', we can adjust this in the Excel file and use `get_metadata()` again.

> We will simulate this behavior here by manually adjusting the metadata file:

Reviewer: It may be worthwhile to show a screenshot of this change in the Excel file. Otherwise, this code is pretty confusing for a newbie.
